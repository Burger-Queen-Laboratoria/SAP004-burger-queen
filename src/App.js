import React from "react";
import firebase from "./init-firebase.js";
import { BrowserRouter, Switch, Route } from "react-router-dom";
import { LoginPage } from "./login/LoginPage.js";
import { LoungePage } from "./lounge/LoungePage.js";
import { About } from "./sobre.js";

console.log(firebase);

function App() {
  return (
    <BrowserRouter>
      <Switch>
        <Route path="/" exact={true} component={LoginPage} />
        <Route path="/lounge" component={LoungePage} />
        <Route path="/sobre" component={About} />
      </Switch>
    </BrowserRouter>
  );
}

export default App;
