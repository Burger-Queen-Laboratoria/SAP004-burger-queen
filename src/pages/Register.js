import React, { useState } from "react";
import { fireFuncs } from "../firebase/firebaseFunctions.js";
import { InputComponent } from "../components/Input.js";
import { Button } from "../components/Button.js";
import { Checkbox } from "../components/Checkbox.js";
import { ErrorDictionary } from "../firebase/error.js";
import {
  BtnsRegisterContainer,
  TitleLogo,
  StyleForm,
  Title,
} from "../components/StyleComponents.js";
import { ErrorArea } from "../components/Errors.js";

export const Register = () => {
  const [name, setName] = useState(""),
    [email, setEmail] = useState(""),
    [password, setPassword] = useState(""),
    [sectorJob, setSectorJob] = useState([]),
    [errorMessage, setErrorMessage] = useState("");

  const setElements = (event, callbackSet) => {
    event.preventDefault();
    callbackSet(event.target.value);
  };

  const createUser = (event) => {
    event.preventDefault();
    fireFuncs
      .authCreateUser(email, password)
      .then((e) => {
        console.log(e);
      })
      .catch((error) => {
        const firebaseError = new ErrorDictionary(error);
        setErrorMessage(firebaseError.translate());
      });
  };

  return (
    <StyleForm>
      <TitleLogo>Preencha os campos abaixo:</TitleLogo>
      <InputComponent
        type="text"
        text="Nome e sobrenome"
        func={(e) => {
          setElements(e, setName);
        }}
      />
      <InputComponent
        type="email"
        text="exemplo@exemplo.com"
        require
        func={(e) => {
          setElements(e, setEmail);
        }}
      />
      <Title>Selecione o setor:</Title>

      <Checkbox
        options={["SalÃ£o", "Cozinha"]}
        value={sectorJob}
        setValue={setSectorJob}
      />

      <InputComponent
        type="password"
        text="senha"
        func={(e) => {
          setElements(e, setPassword);
        }}
      />

      <BtnsRegisterContainer>
        <Button type="submit" name="Registrar" onClick={createUser}></Button>
        <Button type="button" name="Voltar"></Button>
      </BtnsRegisterContainer>
      <ErrorArea err={errorMessage} />
    </StyleForm>
  );
};
