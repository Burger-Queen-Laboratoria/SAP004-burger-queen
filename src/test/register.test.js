import React from "react";
import ReactDOM from "react-dom";
import App from "../App";
import { shallow } from "enzyme";
import Register from "../pages/Register";

it("should display an input to fill question", () => {
  const wrapper = shallow(<Register />);
  expect(wrapper.find("input").length).toBe(3);
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
});
