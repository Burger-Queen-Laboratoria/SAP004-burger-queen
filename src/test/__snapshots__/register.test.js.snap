// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display an input to fill question 1`] = `ReactWrapper {}`;
